src/grid_interpolator.cc

```diff
@@ -0,0 +1,104 @@
+#include "flyft/grid_interpolator.h"
+
+#include <fstream> 
+#include <cassert>
+
+namespace flyft
+{
+GridInterpolator::GridInterpolator(std::string s)
+    :GridInterpolator(s, 0, 0, 0)
+    {
+    // Default constructor for the ThreeDimensionIndex
+    }
+
+GridInterpolator::GridInterpolator(std::string s, int ni, int nj, int nk):
+    n_(ni, nj ,nk)
+    {
+    double dx, dy, dz;
+    std::ifstream indata;
+    indata.open(s);
+    if(!indata.is_open())
+        {
+        std::cout<<"File failed to open"<<std::endl;
+        }
+    indata>>ni>>nj>>nk>>dx>>dy>>dz;
+    indata.close();
+    GridInterpolator(s, ni, nj, nk, dx, dy, dz);
+    }    
+
+GridInterpolator::GridInterpolator(std::string s, int ni, int nj, int nk, double dx, double dy, double dz):
+    n_(ni, nj, nk), dx_(dx), dy_(dy), dz_(dz)
+    {
+    data_ = new double[n_.size()];  
+    std::ifstream indata;
+    indata.open(s);
+    while(indata.good()) 
+        {
```

Comment:
I don't think this is the right way to open & read a file, as far as I remember. I would have done:
```suggestion
    std::ifstream indata(s);
    if (!indata)
        {
        throw std::runtime_error("Failed to open file");
        }
```
Simpler initialization, no while loop needed.